src/ngraph/op/reverse_sequence.hpp

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+* Copyright 2017-2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <memory>
+
+#include "ngraph/op/util/requires_tensor_view_args.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        class ReverseSequence : public util::RequiresTensorViewArgs
+        {
+        public:
+            /// \brief Constructs an arcsin operation.
+            ///
+            /// \param arg Node that produces the input tensor.
+            ReverseSequence(const std::shared_ptr<Node> arg,
+                            size_t batch_axis,
+                            size_t seq_axis,
+                            const Shape& seq_lengths);
```

Comment:
the sequence lengths needs to be a Node, we chatted about it in person.

Reply:
And an attribute the specifies the axis that is lengthed, and the axis that indexes the node length.